### ISSUE TITLE
Viz: Merge Bar/Violin/box plots and superplots

### ISSUE THREAD

**Issue by ChristinaSchmidt1** [saezlab/MetaProViz#30]
As discussed yesterday please combine these four functions into one:
1. Bargraph
2. Boxplot
3. Violinplot
4. superplots

Give the user the parameter GraphStyle= "Bar", "Box" or "Violin"
Give the user the parameter Superplots = TRUE or FALSE (if TRUE the user needs to provide the column name they want to use to colour code the superplots for.

**Comment by dprymidis** [comment]
You call the vizSuperplots like this

VizSuperplot(Input_data = Intra_Preprocessed[,-c(1:3, 30:182)],
             Input_SettingsFile = Intra_Preprocessed[,c(1:2)],
             Input_SettingsInfo = c(conditions="Conditions", superplot = "Biological_Replicates"),
             Graph_Style = "Box", # Bar, Box, Violin
            # Superplot = NULL,
             OutputPlotName = "",
             Individual_plots = TRUE,
             Selected_Conditions = c("786-M1A", "786-O", "HK2"), 
             Selected_Comparisons = list(c(1,2), c(1,3), c(2,3)),
             Theme = theme_classic(),
             Save_as_Plot = "svg") # for together it always pdf

Now about Selected_Conditions . if NULL then all groups are plotted. If some are selected then only those are plotted with the same order as  in the  Selected_Conditions  vector. 
The Selected_Comparisons. If NULL then no stats are added. if one pair is added then t.test, if more than 1 pairs are added then anova

**Comment by dprymidis** [comment]
I did not double check if everything is working as it should with no problems. But yes, you can start on the helper function

**Comment by ChristinaSchmidt1** [comment]
Ok I had a look at the function and tested the different functionalities. A couple of points:

- [ ] Add error bars to violin and bar graphs
- [ ] rename parameters to be in line with other visuals (Plot_Settingsfile, Plot_SettingsInfo, color_palette)
- [ ] When I set Individual=FALSE, the produced PDF is deprecated
- [ ] Stats are AOV at the moment. lets enable the different stats as in DMA (t-test, wilcoxon.test, AOV, kruskal). Do not allow t.test when there is more than 2 conditions. 

The above are the things I noticed when going trough the function thus far.
